tests/test_unit.py

```diff
@@ -54,3 +65,200 @@ def test_repeating_parameters(add_duplicate):
+        (InputProdList, [1, 2, 3], {'_format': 'names_list'}, [1, 2, 3]),
+        (InputProdList, [1, 2, 3], {}, [1, 2, 3]),
+        (InputProdList, [1, 2, 3], {'_format': 'things_list'}, RuntimeError),
+        (InputProdList, {}, {}, RuntimeError),
+        (InputProdList, 1, {'_format': 'names_list'}, ['1']),
+        (InputProdList, 'aaa', {'_format': 'names_list'}, ['aaa']),
+        (InputProdList, '1 2 34', {'_format': 'names_list'}, ['1 2 34']),
+        (InputProdList, [1, '2', 3], {'_format': 'names_list'}, [1, '2', 3]),
+    ]:
+        def constructor():
+            return parameter_type(value=input_value,
+                                  name="my-parameter-name",
+                                  **format_args
+                                  )
+        if isinstance(outcome, type) and issubclass(outcome, Exception):
+            with pytest.raises(outcome):
+                constructor()
+        else:
+            # this also sets the default value
+            parameter = constructor()
+            # check stuff on the inputProdList
+            assert parameter.value == outcome
+
+
+@pytest.mark.parametrize("value",  [25, 25., 25.64547871216879451687311211245117852145229614585985498212321,
+                                    "25", "25.", "25.64547871216879451687311211245117852145229614585985498212321",
+                                    "aaaa"])
+def test_float_defaults(value):
+    if isinstance(value, str) and not value.replace('.', '').isdigit():
+        with pytest.raises(RuntimeError):
+            Float(
+                value=value,
+                name="p_float"
+            )
+    else:
+        p_float = Float(
+            value=value,
+            name="p_float"
+        )
+
+        assert p_float.get_value_in_default_format() == float(value)
+        assert type(p_float.value) == float
+
+
+@pytest.mark.parametrize("e_units", ['eV', 'W', '', None])
+def test_spectral_boundaries_defaults(e_units):
+    # test with a not allowed unit
+    if e_units == 'W':
+        with pytest.raises(RuntimeError):
+            SpectralBoundary(
+                value=10.,
+                name="p_spectral_boundary",
+                E_units=e_units,
+            )
+    else:
+        p_spectral_boundary = SpectralBoundary(
+            value=10.,
+            name="p_spectral_boundary",
+            E_units=e_units,
+        )
+
+        assert p_spectral_boundary.get_value_in_default_format() == p_spectral_boundary.value
+        assert p_spectral_boundary.get_value_in_default_format() == float(10.)
+        assert type(p_spectral_boundary.value) == float
+
+
+@pytest.mark.parametrize("value",  [25, 25., 25.64547871216879451687311211245117852145229614585985498212321,
+                                    "25", "25.", "25.64547871216879451687311211245117852145229614585985498212321",
+                                    "aaaa"])
+def test_integer_defaults(value):
```

Comment:
I currently see no difference between `float` and `int` which can not be absorbed in one test. See other suggested PR.

Reply:
yes I will remove it,